src/System.Management.Automation/engine/CoreAdapter.cs

```diff
@@ -2585,9 +2624,33 @@ internal MethodCacheEntry(MethodBase[] methods)
                     return methodInformationStructures[i];
                 }
             }
+
+            private bool? _isHidden;
+            internal override bool IsHidden
+            {
+                get
+                {
+                    if (_isHidden == null)
```

Comment:
If multiple threads access this property, we'll do the work to discover the hidden attribute multiple times. But, I don't believe that would cause a bug, just does needless work. OTOH, the easiest way to make this threadsafe (`Lazy<bool>(IsMemberHidden)`) probably adds some overhead on both generation and access (or we could just add a lock per instance I suppose). So I'm guessing the solution here sells out an uncommon bad case with race conditions (which aren't really race conditions, since the result is always the same) for the usual case?

Reply:
I see we lock cache tables but not cache elements in the whole file.

Reply:
To me, this is more of a small optimization so that we don't have to calculate the value every time this property gets accessed. Adding a lock is expensive and not needed in this case given that the operation is so simple that it's OK for multiple threads run in the `if (_isHidden == null)` block occasionally.
Previously, the `ishidden` check happens in higher level like `GetDotNetPropertyImpl` and `GetDotNetMethodImpl`, and it happens every time when those methods called. So having this small optimization is worthwhile.

Reply:
I agree, just wanted to make sure we discussed it

Reply:
-        /// Get the first no-hidden member that satisfies the predicate.

+        /// Get the first non-hidden member that satisfies the predicate.